CircleSpace/CircleSpaceServiceLib/Service/SqlCricleSpaceService.cs

```diff
@@ -0,0 +1,361 @@
+        {
+            List<string> list = new List<string>();
+            var images = p.Images.ToList();
+            images.ForEach(image =>
+            {
+                list.Add(image.ImageName);
+            });
+            return list;
+        }
+        //private ICollection<Image> ImageUrlsToImages(PageModel p)
+        //{
+        //    ICollection<Image> list = new List<Image>();
+        //    var images = p.Images.ToList();
+        //    images.ForEach(image =>
+        //    {
+        //        list.Add(image.ImageName);
+        //    });
+        //    return list;
+        //}
+
+        private UserModel UserToUserModels(User u)
+        {
+            UserModel newUser = new UserModel()
+            {
+                ID = u.ID,
+                Username = u.Username,
+                FirstName = u.FirstName,
+                LastName = u.LastName,
+                Email = u.Email
+
+            };
+            return newUser;
+        }
+
+        private List<UserModel> UsersToContributors(Page p)
+        {
+            List<UserModel> contributors = new List<UserModel>();
+            var users = p.Users.ToList();
+            users.ForEach(user =>
+            {
+                contributors.Add(UserToUserModels(user));
+            });
+            return contributors;
+        }
+
+        private List<LayoutModel> LayoutsToListOfLayouts(IEnumerable<Layout> list)
+        {
+            List<LayoutModel> result = new List<LayoutModel>();
+            list.ToList().ForEach(layout =>
+            {
+                var l = LayoutToLayoutModel(layout);
+                result.Add(l);
+            });
+            return result;
+
+        }
+        private LayoutModel LayoutToLayoutModel(Layout l)
+        {
+            LayoutModel newLayout = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                newLayout = new LayoutModel()
+                {
+                    ID = l.ID,
+                    Owner = UserToUserModels(db.Users.Where(u => u.ID == l.OwnerID).First()),
+                    LayoutTitle = l.LayoutTitle,
+                    Content = l.Content,
+                    CSS = l.CSS,
+                    Type = (LayoutTypes)Enum.Parse(typeof(LayoutTypes), l.LayoutType)
+                };
+                var tags = l.Tags.ToList();
+                tags.ForEach(tag =>
+                {
+                    newLayout.Tags.Add(tag.Tag1);
+                });
+            }
+
+            return newLayout;
+
+        }
+
+        private Layout LayoutModelToLayout(LayoutModel l)
+        {
+            Layout newLayout = new Layout()
+            {
+                ID = l.ID,
+                LayoutTitle = l.LayoutTitle,
+                LayoutType = l.Type.ToString(),
+                Content = l.Content,
+                CSS = l.CSS,
+
+            };
+            return newLayout;
+        }
+    }
```

Comment:
Add this nested class in the same file    

private class TagNameComparer : IEqualityComparer<Tag>
        {
            public bool Equals(Tag x, Tag y)
            {
                return x.Tag1 == y.Tag1;
            }

            public int GetHashCode(Tag obj)
            {
                throw new NotImplementedException();
            }
        }